frontend/TvCv19.Frontend/ClientApp/projects/shared-caregiver/src/lib/physician-checklist/physician-checklist.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit, EventEmitter, Input } from '@angular/core';
+import { MatCheckboxModule } from '@angular/material/checkbox';
+import { MatListModule } from '@angular/material/list';
+import { ActivatedRoute, Router } from '@angular/router';
+import { PhysicianChecklistModel } from '../physician-checklist-model';
+import { ToolbarService } from 'src/app/toolbar.service';
+import {
+  PhysicianService,
+  PatientService,
+} from 'projects/shared/src/public-api';
+
+@Component({
+  selector: 'lib-physician-checklist',
+  templateUrl: './physician-checklist.component.html',
+  styleUrls: ['./physician-checklist.component.scss'],
+})
+export class PhysicianChecklistComponent implements OnInit {
+  patientId: string;
+  physicianId: string;
+  patientName: string;
+  physicianName: string;
+  uncheckedWarnText: boolean;
+  checkListForm;
+
+  inputItems = [
+    'Check Manual resuscitator',
+    'Check Suction',
+    'Check Other Items',
+    'Listen and document breath sounds',
+    'Check ETT(endotracheal tube) position',
+    'Compare ETT to previous assessment',
+    'Check Inline suction catheter',
+    'If need: Change Inline suction catheter',
+    'Check HME condition ',
+    'Change HME if needed',
+    'Check connections tight and secure',
+    'Check if humidifier chamber filling',
+    'Check humidifier H2O source sufficient',
+  ];
+
+  checkList: Array<PhysicianChecklistModel> = [];
+
+  constructor(
+    private toolbar: ToolbarService,
+    private physicianService: PhysicianService,
+    private patientService: PatientService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {
+    //get the patientId
+    route.parent.params.subscribe((p: any) => {
+      this.patientId = p['id'];
+    });
+
+    //get the physicianId
+    route.parent.parent.params.subscribe((p) => {
+      this.physicianId = p['id'];
+    });
+
+    //try to grab the patient's name using patient service and set toolbar
+    this.patientService
```

Comment:
this could be moved into line 52 if you are not using the patient id for anything
then you could make the subscribe async and await the response from the get patient

Reply:
moved over